### PR TITLE
native python documentation on RTD

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

File: python/amici/ode_export.py [hunk 1]
```diff
@@ -247,49 +252,46 @@
...
-    Arguments:
-
-        name: name of the function @type str
-
-        varname: name of the symbolic variable @type str
-
-    Returns:
-
-    Raises:
+    :param name:
```
Comment (dweindl) [inline python/amici/ode_export.py]: I think I liked the old comment formatting better for reading in-source. :grimacing:

Comment (FFroehlich) [reply python/amici/ode_export.py]: Thats the format that swig generates though and the formatting that sphinx generates is different for these formats

Comment (FFroehlich) [reply python/amici/ode_export.py]: Also pyment can interconvert docstring formats, so switching to a different supported format such as numpydoc shouldn’t be such a big problem.

Comment (dweindl) [reply python/amici/ode_export.py]: Agreeing that it is preferable to have a consistent style. This one is also better supported by e.g. pycharm. Also RST allows for some other nice features. Still finding it rather ugly :). Wasn't aware, that sphinx produces different results of it. Used the old style also e.g. in PEtab, and things looked well after sphinx. Never compared.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

File: python/amici/ode_export.py (hunk 1 above)
Comment (FFroehlich) [reply python/amici/ode_export.py]: Could also have been me not properly setting up sphinx at first :D
Lets see what things look like when this is done, I think most of the refactoring should be done :)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
not sure what the best approach is to actually get this to RTD, looks like there isn't any option for manual upload/deployment?

Alternative would be to also deploy this to github pages, but this would only give us the doc for the latest commit on master.

**Comment by FFroehlich** [comment]
> not sure what the best approach is to actually get this to RTD, looks like there isn't any option for manual upload/deployment?

https://github.com/readthedocs/readthedocs.org/issues/1083 🙄

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

File: documentation/conf.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+
+# -- Options for Texinfo output ----------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (master_doc, 'AMICI', 'AMICI Documentation',
+     author, 'AMICI', 'One line description of project.',
```
Comment (dweindl) [inline documentation/conf.py]:
```suggestion
     author, 'AMICI', 'Advanced Multilanguage Interface for CVODES and IDAS.',
```

File: include/amici/abstract_model.h [hunk 3]
```diff
@@ -48,8 +48,7 @@ class AbstractModel {
      * @param t time
      * @param x state
      * @param dx time derivative of state (DAE only)
-     * @param xdot array to which values of the residual function will be
-     * written
+     * @param xdot array to which values of the residual function will be written
```
Comment (dweindl) [inline include/amici/abstract_model.h]: Intended change in line width? I liked the 80... If intended, please update in `.clang-format`.

File: python/amici/__init__.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
-""" @package amici
-The AMICI Python module (in doxygen this will also contain documentation about the C++ library)
-
-The AMICI Python module provides functionality for importing SBML models and turning them into C++ Python extensions.
+"""
+AMICI
+------
```
Comment (dweindl) [inline python/amici/__init__.py]:
```suggestion
-----
```

File: python/amici/gradient_check.py [hunk 5]
```diff
@@ -1,11 +1,64 @@
...
+                            solver: Solver,
+                            edata: ExpData,
+                            ip: int,
+                            fields: List[str],
+                            assert_fun: Callable,
+                            atol: Optional[float] = 1e-12,
+                            rtol: Optional[float] = 1e-12,
+                            epsilon: Optional[float] = 1e-4) -> None:
+    """
+    Checks the computed sensitity based derivatives against a finite
```
Comment (dweindl) [inline python/amici/gradient_check.py]:
```suggestion
    Checks the computed sensitivity based derivatives against a finite
```

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

File: python/amici/logging.py [hunk 6]
```diff
@@ -88,31 +100,39 @@ def _setup_logger(level=logging.WARNING, console_output=True,
...
 
 
-def get_logger(logger_name=BASE_LOGGER_NAME, log_level=None, **kwargs):
-    """ Returns (if extistant) or creates an AMICI logger
+def get_logger(logger_name: Optional[str] = BASE_LOGGER_NAME,
+               log_level: Optional[int] = None,
+               **kwargs) -> logging.Logger:
+    """
+    Returns (if extistant) or creates an AMICI logger
     If the AMICI base logger has already been set up, this method will
```
Comment (dweindl) [inline python/amici/logging.py]:
```suggestion

    If the AMICI base logger has already been set up, this method will
```

File: python/amici/ode_export.py [hunk 7]
```diff
@@ -1,7 +1,11 @@
-""" @package amici.ode_export The C++ ODE export module for python
 """
-#!/usr/bin/env python3
-
+C++ Export
+------------
```
Comment (dweindl) [inline python/amici/ode_export.py]:
```suggestion
----------
```

File: python/amici/petab_import.py [hunk 8]
```diff
@@ -1,4 +1,6 @@
 """
+PEtab Import
+---------------
```
Comment (dweindl) [inline python/amici/petab_import.py]:
```suggestion
------------
```

File: scripts/run-sphinx.sh [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+AMICI_PATH=$(cd $SCRIPT_PATH/.. && pwd)
+
+cd ${AMICI_PATH}/documentation
+source ../build/venv/bin/activate
+pip3 install sphinx nbsphinx recommonmark sphinx_rtd_theme petab sphinx-autodoc-typehints
+
+# add linebreaks that swig is missing, we can repeat this as many times as
+# we want
+AMICI_FILE=${AMICI_PATH}/python/sdist/amici/amici.py
+sed -i -e -E $'s/([ ]+):(rtype|type|param|return)/\\\n\\1:\\2/g' $AMICI_FILE
```
Comment (dweindl) [inline scripts/run-sphinx.sh]: What went wrong there? :)

File: python/amici/sbml_import.py [hunk 10]
```diff
@@ -1,4 +1,10 @@
-"""@package amici.sbml_import The SBML import module for Python."""
+"""
+SBML Import
+------------
```
Comment (dweindl) [inline python/amici/sbml_import.py]:
```suggestion
-----------
```

**Comment by dweindl** [comment]
> not sure what the best approach is to actually get this to RTD, looks like there isn't any option for manual upload/deployment?

Do we need to upload the finished thing? We can also generate everything at RTD, can't we?

File: scripts/run-sphinx.sh (hunk 9 above)
Comment (FFroehlich) [reply scripts/run-sphinx.sh]: swig conversion of docstrings is not as good as you want and seemingly arbitrarily misses linebreaks from time to time 🤷‍♂

**Comment by FFroehlich** [comment]
> > not sure what the best approach is to actually get this to RTD, looks like there isn't any option for manual upload/deployment?
> 
> Do we need to upload the finished thing? We can also generate everything at RTD, can't we?

That means we need to build amici on RTD, right? I don't see a good way of installing BLAS & swig there.

**Comment by dweindl** [comment]
> That means we need to build amici on RTD, right? I don't see a good way of installing BLAS & swig there.

Yes. Would also have to read up the details, but I think it's feasible (https://docs.readthedocs.io/en/stable/builds.html). BLAS could probably be circumvented somehow, as we don't need to compile anything, but only convince SWIG to generate the docstrings.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
> > That means we need to build amici on RTD, right? I don't see a good way of installing BLAS & swig there.
> 
> Yes. Would also have to read up the details, but I think it's feasible (https://docs.readthedocs.io/en/stable/builds.html). BLAS could probably be circumvented somehow, as we don't need to compile anything, but only convince SWIG to generate the docstrings.

Looks like they recently added swig to their docker images, so this might actually work

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
> > > That means we need to build amici on RTD, right? I don't see a good way of installing BLAS & swig there.
> > 
> > 
> > Yes. Would also have to read up the details, but I think it's feasible (https://docs.readthedocs.io/en/stable/builds.html). BLAS could probably be circumvented somehow, as we don't need to compile anything, but only convince SWIG to generate the docstrings.
> 
> Looks like they recently added swig to their docker images, so this might actually work:
> [readthedocs/readthedocs-docker-images#15](https://github.com/readthedocs/readthedocs-docker-images/issues/15)

But its swig3.0, so we won't get the sphinx compatible typehints. I don't think they will support swig4.0 anytime soon, so we better look for other options. With this, I currently don't see any way of directly using RTD since they neither support manual upload, nor custom docker images.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  6
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
> > That means we need to build amici on RTD, right? I don't see a good way of installing BLAS & swig there.
> 
> Yes. Would also have to read up the details, but I think it's feasible (https://docs.readthedocs.io/en/stable/builds.html). BLAS could probably be circumvented somehow, as we don't need to compile anything, but only convince SWIG to generate the docstrings.

Why don't we need to compile anything? The modules are imported by swig after all ...
I managed to implement a custom RTD build routing from the `conf.py`, but as expected BLAS seems to be a problem: https://readthedocs.org/projects/amici/builds/10528892/

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  8
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  9
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by dweindl** [comment]
> But its swig3.0, so we won't get the sphinx compatible docstrings.

We can't simply execute our swig4 install script from `conf.py`? 

> Why don't we need to compile anything? The modules are imported by swig after all ...

By sphinx? Could probably be tweaked that we don't need the extension to load, but that the python wrapper is enough. The documentation is included there, not in the shared object.

> I managed to implement a custom RTD build routing from the `conf.py`, but as expected BLAS seems to be a problem: https://readthedocs.org/projects/amici/builds/10528892/

Could use some dummy files, or as for the first point, we can try installing it ourselves.

**Comment by dweindl** [comment]
Want to merge the current state already and test deployment afterwards? I see much potential for merge conflicts :)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
> Want to merge the current state already and test deployment afterwards? I see much potential for merge conflicts :)

had some limited success with mock modules, but I agree, will merge as soon as test are done.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Comment by FFroehlich** [comment]
actually managed to fix it 🎉 
https://amici.readthedocs.io/en/readthedocs/modules.html

**Comment by FFroehlich** [comment]
(almost)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- documentation/conf.py  15
         

Clones removed
==============
+ python/amici/pandas.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5009562&bid=16524218)

**Pull request merged**